api/app/db/crud/auto_spatial_advisory.py

```diff
@@ -386,3 +423,29 @@ async def get_provincial_rollup(session: AsyncSession, run_type: RunTypeEnum, ru
     )
     result = await session.execute(stmt)
     return result.all()
+
+
+async def get_containing_zone(session: AsyncSession, geometry: str, srid: int):
+    geom = func.ST_GeomFromText(geometry, srid)
+    stmt = select(Shape.id).filter(func.ST_Contains(Shape.geom, geom))
```

Comment:
I think there might be a potential projection issue here if `srid` isn't the same projection as `Shape`'s srid. Could possibly make it more generic with an `ST_Transform`

Reply:
You mean something like this?

`geom = func.ST_Transform(func.ST_GeomFromText(geometry, srid), 3005)`

Since 3005 is what we're storing in `advisory_shapes`.

Reply:
Ya, could even do the transform conditionally if srid != 3005 🤷🏻‍♂️

Reply:
If we're going to do the transform here (which is a great idea), I think we can get rid of the `PointTransformer` class in `geospatial.py`.